packages/notification-services-controller/src/NotificationServicesController/types/notification/notification.ts

```diff
@@ -1,10 +1,12 @@
 import type { FeatureAnnouncementRawNotification } from '../feature-announcement/feature-announcement';
 import type { OnChainRawNotification } from '../on-chain-notification/on-chain-notification';
+import type { RawSnapNotification } from '../snaps';
 import type { Compute } from '../type-utils';
 
 export type NotificationUnion =
   | FeatureAnnouncementRawNotification
-  | OnChainRawNotification;
+  | OnChainRawNotification
+  | RawSnapNotification;
```

Comment:
Nice!

We do see below that the INotification below does not handle the `readDate` expected in a normalised snap notification.

Although I like type-fu, lets modify INotification to be a bit more expressive.
```ts
type BaseNotification = {
  id: string;
  createdAt: string
  isRead: boolean
}

export type INotification = Compute<
  | FeatureAnnouncementRawNotification & BaseNotification
  | OnChainRawNotification & BaseNotification
  | RawSnapNotification & BaseNotification & { readDate?: Date | null }
>

```